pyowc/configuration.py

```diff
@@ -0,0 +1,254 @@
+        'FLOAT_CALIB_PREFIX': "cal_",
+        'FLOAT_CALSERIES_PREFIX': "calseries_",
+        'FLOAT_CALIB_POSTFIX': ".mat",
+
+        # ===============================
+        #
+        #    Diagnostic Plots Output Path
+        #
+
+        'FLOAT_PLOTS_DIRECTORY': "data/float_plots/",
+
+        # ===============================
+        #
+        #    Constants File Path
+        #
+
+        'CONFIG_DIRECTORY': "data/constants/",
+        'CONFIG_COASTLINES': "coastdat.mat",
+        'CONFIG_WMO_BOXES': "wmo_boxes.mat",
+        'CONFIG_SAF': "TypicalProfileAroundSAF.mat",
+
+        # ===============================
+        #
+        #    Objective Mapping Parameters
+        #
+
+        # max number of historical casts used in objective mapping
+        'CONFIG_MAX_CASTS': 300,
+
+        # 1=use PV constraint, 0=don't use PV constraint, in objective mapping
+        'MAP_USE_PV': 0,
+
+        # 1=use SAF separation criteria, 0=don't use SAF separation criteria, in objective mapping
+        'MAP_USE_SAF': 0,
+
+        # spatial decorrelation scales, in degrees
+        'MAPSCALE_LONGITUDE_LARGE': 8,
+        'MAPSCALE_LONGITUDE_SMALL': 4,
+        'MAPSCALE_LATITUDE_LARGE': 4,
+        'MAPSCALE_LATITUDE_SMALL': 2,
+
+        # cross-isobath scales, dimensionless, see BS(2005)
+        'MAPSCALE_PHI_LARGE': 0.5,
+        'MAPSCALE_PHI_SMALL': 0.1,
+
+        # temporal decorrelation scale, in years
+        'MAPSCALE_AGE_LARGE': 20,
+        'MAPSCALE_AGE_SMALL': 10,
+
+        # exclude the top xxx dbar of the water column
+        'MAP_P_EXCLUDE': 200,
+
+        # only use historical data that are within +/- yyy dbar from float data
+        'MAP_P_DELTA': 250
+    }
+    return collections.OrderedDict(sorted(cfg.items()))
+
+
+#pylint: disable=too-many-locals
+#pylint: disable=too-many-branches
+#pylint: disable=too-many-statements
+#pylint: disable=too-many-nested-blocks
+#pylint: disable=invalid-name
+#pylint: disable=fixme
+def set_calseries(float_dir, float_name, system_config):
+    """ Set the calseries parameters for analysis and line fitting
+
+        Parameters
+        ----------
+        use_pres_gt: pressure greater than
+        use_theta_gt: ptmp greater than
+        use_theta_lt: ptmp less than
+        use_pres_lt: pressure less than
+        float_dir: location of float
+        float_name: float source name
+        system_config: configuration settings
+
+        Returns
+        -------
+        Nothing, but save parameters
+    """
+
+    # load float source data
+    float_source = loadmat(os.path.sep.join([system_config['FLOAT_SOURCE_DIRECTORY'], float_dir,
+                                             float_name + system_config['FLOAT_SOURCE_POSTFIX']]))
+
+    profile_no = float_source['PROFILE_NO'].flatten()
+    no_profiles = profile_no.__len__()
+
+    # Check if we already have a calseries file
+
+    calseries_filename = (os.path.sep.join([system_config['FLOAT_CALIB_DIRECTORY'], float_dir,
```

Comment:
There's something odd going on here.` calseries_filename` is being set as `data/float_calib///calseries_3901960.mat`, and whilst this does work it looks odd.

Despite this though, the individual variables are being set correctly, ie:

- `system_config['FLOAT_CALIB_DIRECTORY'] = data/float_calib`
- `float directory = /`
- `float_name = 3901960`
- `system_config['FLOAT_SOURCE_POSTFIX'] = .mat`

Is `os.path.sep.join` just doing something weird?

Reply:
![image](https://user-images.githubusercontent.com/55086028/87042441-9769f380-c1eb-11ea-99e1-45059c6340fe.png)

![image](https://user-images.githubusercontent.com/55086028/87042414-8f11b880-c1eb-11ea-9864-684db7507cc8.png)

Reply:
``os.path.sep`` gives a "/" or a "\" depending on the platform you are, 
so no issue here, it's like
```python
"/".join(['data/float_calib', '/', '3901960.mat'])
```
producing indeed a weird ``data/float_calib///calseries_3901960.mat``
The real problem is the non-necessary use of the ``float_dir`` parameter when it is only equal to ``os.path.sep``.
I didn't fixed this because it's link to a refactoring of the paths management and configuration.
Basically, we should have a module to manage data access that is responsible for producing the appropriate path and data for a math function.